### ISSUE TITLE
different isoforms asigned

### ISSUE THREAD

**Issue by jjgao** [genome-nexus/genome-nexus#205]
If we submit the mutations (all in HIST1H2BN) below for anntoation, the last one was returned on a different isoform. Is it because the mutation is in UTR and therefore a missense mutation in another isoform was picked regardless which cannonical transcript we are choosing?

http://www.cbioportal.org/beta/mutation_mapper.jsp

Tumor_Sample_Barcode | NCBI_Build | Chromosome | Start_Position | End_Position | Variant_Type | Reference_Allele | Tumor_Seq_Allele1
-- | -- | -- | -- | -- | -- | -- | --
TCGA-UY-A9PB-01 | GRCh37 | 6 | 27806523 | 27806523 | SNP | G | T
TCGA-T2-A6WX-01 | GRCh37 | 6 | 27806511 | 27806511 | SNP | G | T
TCGA-P3-A5Q5-01 | GRCh37 | 6 | 27806606 | 27806606 | SNP | C | T
TCGA-OR-A5JB-01 | GRCh37 | 6 | 27806779 | 27806779 | SNP | G | A
TCGA-GU-A42P-01 | GRCh37 | 6 | 27806606 | 27806606 | SNP | C | T
TCGA-GM-A3NW-01 | GRCh37 | 6 | 27806549 | 27806549 | SNP | G | A
TCGA-FI-A2F4-01 | GRCh37 | 6 | 27806627 | 27806627 | SNP | T | C
TCGA-EO-A22R-01 | GRCh37 | 6 | 27819890 | 27819890 | SNP | A | G

![image](https://user-images.githubusercontent.com/840895/42891797-36bee3e8-8a7e-11e8-833f-513a3e8d7355.png)

It's also happening in the main portal.

http://www.cbioportal.org/index.do?session_id=5b4f5e2d498eb8b3d567026f

Once this is fixed, we may want to reannotate mutations in the portal database and files. We can do a database query to find out the mutations that were not annotated on the cannonical isoforms.

**Comment by inodb** [comment]
@jj	@onur I think there are multiple issues here. One issue is that the annotation-summary endpoint's canonicalTranscriptId is not correct (this is currently only used on beta). It should be the same transcript `ENST00000396980` as in:

http://genomenexus.org/ensembl/canonical-transcript/hgnc/HIST1H2BN?isoformOverrideSource=uniprot

But summary endpoint gives ENST00000606613 as canonicalTranscriptId for HIST1H2BN

https://v2.genomenexus.org/annotation/summary/6%3Ag.27819890A%3EG?isoformOverrideSource=uniprot&projection=ALL

The logic assigns a canonical transcript based on the field canonical (this should actually use the mongo collection we have with assigned canonical transcripts). Because there is none with this field, it then picks the transcript with most severe alteration, missense in `ENST00000606613`. 

So basically there is no consequence in the canonical transcript. In the current logic it picks the most severe one in a different transcript instead. Might be better to just not display it in this case? Maybe we can change summary endpoint to have both fields `canonicalTranscriptId` and `transcriptId`, where the latter represents the transcript id of the most severe alteration in case there is no canonical transcript consequence. Then if these two don't match we don't show anything on the mutation mapper. How should we handle it for `genome-nexus-annotation-pipeline`?
